File: ruby/raindrops/raindrops.rb
class Integer
  def divisible_by?(other)
    self % other == 0
  end
end

module Raindrops
  def self.convert(n)
    output = []

    output << "Pling" if n.divisible_by?(3)
    output << "Plang" if n.divisible_by?(5)
    output << "Plong" if n.divisible_by?(7)

    output << n.to_s if output.empty?

    return output.join
  end
end

File: javascript/raindrops/raindrops.js
export const convert = (n) => {
    let output = [];

    if (n % 3 == 0)
        output.push("Pling");
    if (n % 5 == 0)
        output.push("Plang");
    if (n % 7 == 0)
        output.push("Plong");
    if (output.length == 0)
        output.push(n);

    return output.join('');
};
